src/canvas-extractor/edfi_canvas_extractor/extract_facade.py

```diff
@@ -192,22 +193,30 @@ def run(arguments: MainArguments) -> None:
     if not succeeded:
         _break_execution("Sections")
 
-    succeeded = _get_assignments(arguments, sync_db)
-    if not succeeded:
-        _break_execution("Assignments")
-
     succeeded = _get_students(arguments, sync_db)
     if not succeeded:
         _break_execution("Students")
 
-    _get_system_activities(arguments, sync_db)
-    _get_submissions(arguments, sync_db)
-
     succeeded = _get_enrollments(arguments, sync_db)
     if not succeeded:
         _break_execution("Enrollments")
 
-    _get_grades(arguments, )  # Grades don't need sync process because they are part of enrollments
-    _get_section_activities(arguments, )  # SectionActivities are empty
+    if Features.Assignments in arguments.feature:
+        succeeded = _get_assignments(arguments, sync_db)
+        if succeeded:
+            _get_submissions(arguments, sync_db)
+
+    if Features.Activities in arguments.feature:
+        _get_system_activities(arguments, sync_db)
+        _get_section_activities(
+            arguments,
+        )  # SectionActivities are empty
```

Comment:
I don't understand the comment above. I see that it is an old comment. Do you know what it really means? If not, perhaps we can just delete it?

Reply:
I'll delete it